### ISSUE TITLE
[Bug]: milvus-querycoord && milvus-proxy pod has crashed

### ISSUE THREAD

**Issue by 15820415264** [milvus-io/milvus#19586]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: milvusdb/milvus:v2.1.1
- Deployment mode(standalone or cluster): k8s cluster
- SDK version(e.g. pymilvus v2.0.0rc2): v2.1.0
- OS(Ubuntu or CentOS): ubuntu
- CPU/Memory: 256GB
- GPU: 
- Others:
```

### Current Behavior

1. we have two collections，one have 100 million datas and other one have 31 million datas.
2. The vector dimension of data is 256.
3. we use java sdk to write and query data for test ,and it works well
4. But after sometimes，when we use the milvus client 'ATTU' to see some thing.we get error msg from it.
5. Then we check the services of milvus，we found the milvus-querycoord and milvus-proxy pod has crashed.
![lQLPJxa7xagZO5PNAevNAwOwgGYITLCcMIEDNLZNI4CqAA_771_491](https://user-images.githubusercontent.com/16377504/193209918-e074cf0b-81c2-445c-a63c-836250069ffb.png)
6. The available memory of server is 104GB,but  we can not write and query data any more.

### Expected Behavior

1. how to recovery the milvus services 。
2. The reason of this question.

### Milvus Log

[reboot_my-release-milvus-querycoord.txt](https://github.com/milvus-io/milvus/files/9681594/reboot_my-release-milvus-querycoord.txt)
[my-release-milvus-querycoord-54f7b895d-8l9x8-pre.log](https://github.com/milvus-io/milvus/files/9681595/my-release-milvus-querycoord-54f7b895d-8l9x8-pre.log)
[my-release-milvus-proxy-65566d746c-2ktx2-pre.log](https://github.com/milvus-io/milvus/files/9681596/my-release-milvus-proxy-65566d746c-2ktx2-pre.log)

**Comment by weiliu1031** [comment]
can you offer a etcd backup with birdwather? https://github.com/milvus-io/birdwatcher
/assign @15820415264

**Comment by 15820415264** [comment]
[bw_etcd_ALL.221009-170946.bak.gz](https://github.com/milvus-io/milvus/files/9741388/bw_etcd_ALL.221009-170946.bak.gz)

**Comment by 15820415264** [comment]
[milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/9741389/milvus-log.tar.gz)

**Comment by hukang6677** [comment]
Where is the queryCoord.maxTask？I had a similar problem

**Comment by hukang6677** [comment]
Where is the queryCoord. MaxTask？
I had a similar problem

**Comment by weiliu1031** [comment]
please reference this about how to config milvus: https://milvus.io/docs/v2.1.x/configure-docker.md#Configure-Milvus-with-Docker-Compose-or-Helm

**Comment by weiliu1031** [comment]
> Where is the queryCoord. MaxTask？ I had a similar problem


please reference this about how to config milvus: https://milvus.io/docs/v2.1.x/configure-docker.md#Configure-Milvus-with-Docker-Compose-or-Helm

**Comment by yanliang567** [comment]
@weiliu1031 any solutions to avoid crashes?

**Comment by weiliu1031** [comment]
> @weiliu1031 any solutions to avoid crashes?

reduce active task concurrency may helps! can be config by param `queryCoord. MaxTask`

**Comment by 15820415264** [comment]
> > @weiliu1031 any solutions to avoid crashes?
> 
> reduce active task concurrency may helps! can be config by param `queryCoord. MaxTask`

we didn‘t found 'queryCoord. MaxTask' any where。we try to fix the param 'proxy.MaxTaskNum'  from 1024 to 100 and reboot all pod,but  the pod of 'proxy' and 'queryCoord' still not work 。

**Comment by 15820415264** [comment]
[milvus-log.tar(20221014).gz](https://github.com/milvus-io/milvus/files/9782755/milvus-log.tar.20221014.gz)

**Comment by weiliu1031** [comment]
> MaxTask

try edit `milvus.yaml` like this, if you need.
![image](https://user-images.githubusercontent.com/108661493/195813583-280dfb29-8fc4-4948-9ebb-49a5f69650a1.png)

**Comment by hukang6677** [comment]
k8s helm cluster 
I follow the official document，edit the value.yaml
![image](https://user-images.githubusercontent.com/102129512/196096548-85f2cae0-5de5-422a-90f5-d6df0af4a939.png)
![image](https://user-images.githubusercontent.com/102129512/196097089-8d060823-302d-42d9-b972-61ceebecb12a.png)

#helm upgrade my-release milvus/milvus -f values.yaml

However,the param queryCoord.MaxTask does not work

**Comment by hukang6677** [comment]
I also tried the following operations

#helm upgrade --install my-release milvus/milvus --set queryCoord.MaxTask=100

#helm upgrade --install --set queryCoord.MaxTask=100 my-release milvus/milvus

#helm upgrade --install --set queryCoordinator.MaxTask=100 my-release milvus/milvus

It doesn't work

**Comment by hukang6677** [comment]
the queryCoord.MaxTask=100 doesn't work，I tried the above method

**Comment by weiliu1031** [comment]
> the queryCoord.MaxTask=100 doesn't work，I tried the above method

what do you mean by 'the queryCoord.MaxTask=100 doesn't work' ?  
1. let's make sure that you meet the same question with the origin issue
2. discuss how to solve it

**Comment by weiliu1031** [comment]
> the queryCoord.MaxTask=100 doesn't work，I tried the above method

this param has not been explode to user, so the config method above does not work,  you may need to modify the image, reference: https://phoenixnap.com/kb/how-to-commit-changes-to-docker-image

**Comment by hukang6677** [comment]
![image](https://user-images.githubusercontent.com/102129512/197678402-e5123f5c-3a98-45dd-a896-b168cbf5558a.png)

Into the container
vim milvus.yaml，but milvus.yaml is readonly.
E505: "milvus.yaml" is read-only (add ! to override)

**Comment by weiliu1031** [comment]
> ![image](https://user-images.githubusercontent.com/102129512/197678402-e5123f5c-3a98-45dd-a896-b168cbf5558a.png)
> 
> Into the container vim milvus.yaml，but milvus.yaml is readonly. E505: "milvus.yaml" is read-only (add ! to override)

try another way,  build image base Docker file: https://docs.docker.com/engine/reference/builder

**Comment by hukang6677** [comment]
I use the Dockerfile to generate a new image,and then I use the new image deploy the milvus cluster.but the milvus.yaml  still nothing to be changed

[Dockerfile.txt](https://github.com/milvus-io/milvus/files/9999155/Dockerfile.txt)

**Comment by hukang6677** [comment]
I also modified K8s deployments ,set "readOnly: false", and then exec container edit milvus.yaml ,again to use docker commit , but  doesn't work.
 kubectl edit deployments my-release-milvus-querycoord
![image](https://user-images.githubusercontent.com/102129512/201566281-f05eace2-5c0e-4e16-9377-7b71b3402011.png)

**Comment by weiliu1031** [comment]
> I use the Dockerfile to generate a new image,and then I use the new image deploy the milvus cluster.but the milvus.yaml still nothing to be changed
> 
> [Dockerfile.txt](https://github.com/milvus-io/milvus/files/9999155/Dockerfile.txt)

![image](https://user-images.githubusercontent.com/108661493/201590798-f915a02b-37a3-45a0-aa93-ce73192ca951.png)
please check your step, maybe this will help

**Comment by hukang6677** [comment]
![image](https://user-images.githubusercontent.com/102129512/201808097-572043b1-902c-4f7e-925b-776f9100e8fc.png)

Use your steps，and then use the new image Install Milvus Cluster with Helm (k8s). There is no "maxTask: 100"  in milvus cluster.

![image](https://user-images.githubusercontent.com/102129512/201808801-45b3fc44-bf58-4b65-a2c0-e9b0c8a6f6dd.png)
![image](https://user-images.githubusercontent.com/102129512/201808865-d1b947be-bf08-4619-a1c6-7e50dd959b33.png)

**Comment by weiliu1031** [comment]
> ![image](https://user-images.githubusercontent.com/102129512/201808097-572043b1-902c-4f7e-925b-776f9100e8fc.png)
> 
> Use your steps，and then use the new image Install Milvus Cluster with Helm (k8s). There is no "maxTask: 100" in milvus cluster.
> 
> ![image](https://user-images.githubusercontent.com/102129512/201808801-45b3fc44-bf58-4b65-a2c0-e9b0c8a6f6dd.png) ![image](https://user-images.githubusercontent.com/102129512/201808865-d1b947be-bf08-4619-a1c6-7e50dd959b33.png)
helm will use chart to init a default config file, that's why your changes not work

**Comment by hukang6677** [comment]
"helm will use chart to init a default config file, that's why your changes not work"

You mean to install milvus cluster use other way? 

![image](https://user-images.githubusercontent.com/102129512/201904732-f3dbace2-9161-465a-bff9-924ed50349f6.png)

**Comment by weiliu1031** [comment]
I prefer docker-compose, it's much more simple. otherwise you may need to edit charts(which is used by helm)

**Comment by hukang6677** [comment]
![image](https://user-images.githubusercontent.com/102129512/201912548-fc6972fb-3e56-4049-a8c4-0f4ca5133125.png)

**Comment by hukang6677** [comment]
Our usage scenario is to deploy clusters on multiple servers，so docker-compose may unsuited

**Comment by weiliu1031** [comment]
> Our usage scenario is to deploy clusters on multiple servers，so docker-compose may unsuited

for now, milvus does not support change internal config items,  and work around method doesn't fit your case.

**Comment by yanliang567** [comment]
@weiliu1031 did we fix this issue in 2.2? I think we shall limit the rate or report an error instead of crash.
